### DISCUSSION TITLE
Un necessary compulsion/dependence on RECORD_AUDIO permission

### DISCUSSION THREAD

**Discussion by MTayyabSarfraz** [pedroSG94/RootEncoder#1154] [Q&A]
My project required no audio/video recordings so I omitted the unwanted permissions + codes,
But when I tried to remove `android.permission.RECORD_AUDIO` permission I  found a lot of dependence on this permission, simply omitting permission doesn't work for me (had dependence in **audio Encoders/manager** classes).

Wouldn't it be better to decouple it from the Audio manager/encoder based on user preferences?

**Comment by pedroSG94** [comment]
Hello,

RECORD_AUDIO permission is required if you use AudioRecord class (It is used in MicrophoneManager class of this library)
I'm not sure about your code implemented or your requirements but if you never use MicrophoneManager class you don't need that permission.
Can you show me error logcat?
